NextcloudTalk/UserProfileTableViewControllerUISetup.swift

```diff
@@ -0,0 +1,258 @@
+        let federatedOption = setupDetailedOption(identifier: kUserProfileScopeFederated, imageName: "group", title: "Federated",
+                                                  subtitle: "Only synchronize to trusted servers", selected: currentValue == kUserProfileScopeFederated)
+        let publishedOption = setupDetailedOption(identifier: kUserProfileScopePublished, imageName: "browser-settings", title: "Published",
+                                                  subtitle: "Synchronize to trusted servers and the global and public address book", selected: currentValue == kUserProfileScopePublished)
+        if field != kUserProfileDisplayNameScope && field != kUserProfileEmailScope {
+            options.append(privateOption)
+        }
+        options.append(localOption)
+        let serverCapabilities = NCDatabaseManager.sharedInstance().serverCapabilities(forAccountId: account.accountId)
+        if serverCapabilities.accountPropertyScopesFederationEnabled {
+            options.append(federatedOption)
+            options.append(publishedOption)
+        }
+        let optionSelectorVC = DetailedOptionsSelectorTableViewController(options: options, forSenderIdentifier: field, andTitle: title)
+        if let optionSelectorVC = optionSelectorVC {
+            optionSelectorVC.delegate = self
+            let optionSelectorNC = UINavigationController(rootViewController: optionSelectorVC)
+            self.present(optionSelectorNC, animated: true, completion: nil)
+        }
+    }
+
+    func setUserProfileField(_ field: String?, scopeValue scope: String?) {
+        setModifyingProfileUI()
+        NCAPIController.sharedInstance().setUserProfileField(field, withValue: scope, for: account) { [self] error, _ in
+            if error != nil {
+                showScopeModificationError()
+            } else {
+                refreshUserProfile()
+            }
+            removeModifyingProfileUI()
+        }
+    }
+
+    func showScopeModificationError() {
+        let errorDialog = UIAlertController(title: NSLocalizedString("An error occurred changing privacy setting", comment: ""), message: nil, preferredStyle: .alert)
+        let okAction = UIAlertAction(title: NSLocalizedString("OK", comment: ""), style: .default, handler: nil)
+        errorDialog.addAction(okAction)
+        self.present(errorDialog, animated: true, completion: nil)
+    }
+
+    // Setup Dialog Options
+    func setupDetailedOption(identifier: String, imageName: String, title: String, subtitle: String, selected: Bool) -> DetailedOption {
+        let detailedOption = DetailedOption()
+        detailedOption.identifier = identifier
+        detailedOption.image = UIImage(named: imageName)?.withRenderingMode(.alwaysTemplate)
+        detailedOption.title = NSLocalizedString(title, comment: "")
+        detailedOption.subtitle = NSLocalizedString(subtitle, comment: "")
```

Comment:
Not sure if I'm a fan of localizing the strings given as a parameter. I understand the reasoning for doing it, like it's done, but it makes it hard to check if all strings are localized and also there's no chance of adding a comment for the translators. Any opinions?